meson.build

```diff
@@ -0,0 +1,470 @@
+project('xdg-desktop-portal', 'c',
+         version: '1.13.0',
+         license: 'LGPLv2+',
+         default_options: ['warning_level=2'],
+         meson_version: '>= 0.59.0')
+
+###### various directories we'll use later
+# dir_foodir are built-in ones, dir_foo are our options
+
+dir_prefix = get_option('prefix')
+dir_datadir = get_option('datadir')
+dir_libexecdir = get_option('libexecdir')
+dir_localedir = get_option('localedir')
+dir_dbus_service = get_option('dbus-service-dir')
+if dir_dbus_service == ''
+    dir_dbus_service = dir_datadir / 'dbus-1' / 'services'
+endif
+
+dir_flatpak_intf = get_option('flatpak-interfaces-dir')
+if dir_flatpak_intf == ''
+    dep_flatpak = dependency('flatpak', version: '>= 1.5.0')
+    dir_flatpak_intf = dep_flatpak.get_variable('interfaces_dir')
+endif
+
+dir_systemd_userunit = get_option('systemd-user-unit-dir')
+if dir_systemd_userunit == ''
+    # This is deliberately not ${libdir}: systemd units always go in
+    # .../lib, never .../lib64 or .../lib/x86_64-linux-gnu
+    dir_systemd_userunit = dir_prefix / 'lib' / 'systemd' / 'user'
+endif
+
+dir_dataroot = get_option('datarootdir')
+if dir_dataroot == ''
+    dir_dataroot = dir_datadir
+endif
+
+###### various include directories we'll use later
+# These are set here so meson handles the relative paths correctly,
+# makes life easier for us
+
+inc_builddir = include_directories('.')  # config.h
+inc_src = include_directories('src')
```

Comment:
This naming is misleading: `include_directories('.')` is the Meson equivalent of `-I$(top_builddir) -I$(top_srcdir)`, and `include_directories('src')` is the equivalent of `-I$(top_builddir)/src -I$(top_srcdir)/src`.